ores/wsgi/util.py

```diff
@@ -35,23 +37,147 @@ def read_bar_split_param(request, param, default=None, type=str):
                          .format(param, str(e)))
 
 
+def format_error(error):
+    error_type = error.__class__.__name__
+    message = str(error)
+
+    return {'error': {'type': error_type, 'message': message}}
+
+
+def build_score_request(scoring_system, request, context_name=None, rev_id=None,
+                        model_name=None):
+    rev_ids = parse_rev_ids(request, rev_id)
+    model_names = parse_model_names(request, model_name)
+    precache = 'precache' in request.args
+    include_features = 'features' in request.args
+    injection_caches = parse_injection(request, rev_id)
+    model_info = parse_model_info(request)
+
+    if context_name and context_name in scoring_system and not model_names:
+        model_names = scoring_system[context_name].keys()
+
+    return ScoreRequest(context_name, rev_ids, model_names,
+                        precache=precache,
+                        include_features=include_features,
+                        injection_caches=injection_caches,
+                        model_info=model_info)
+
+
+def parse_rev_ids(request, rev_id):
```

Comment:
Should we do this in a new module, request_parse?  Fine to get there iteratively.